api/app/Actions/Auth/SendLinkForgotPasswordAction.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Actions\Auth;
+
+use App\Exceptions\User\UserNotFoundException;
+use App\Repositories\User\UserRepository;
+use Illuminate\Auth\Notifications\ResetPassword;
+use Illuminate\Notifications\Messages\MailMessage;
+use Illuminate\Support\Facades\Lang;
+use Illuminate\Support\Facades\Password;
+
+final class SendLinkForgotPasswordAction
+{
+    private UserRepository $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+    public function execute(SendLinkForgotPasswordRequest $request)
+    {
+        if (!$this->userRepository->getByEmail($request->getEmail())) {
+            throw new UserNotFoundException();
+        }
+        ResetPassword::toMailUsing(function ($notifiable, $token) {
+            $linkReset = $this->resetUrl($notifiable, $token);
+            $count = config('auth.passwords.' . config('auth.defaults.passwords') . '.expire');
+            return (new MailMessage())
+                ->subject(Lang::get('Reset Password from Schedulia'))
+                ->view('emails.sendLinkResetPassword', ['linkReset' =>$linkReset, 'count'=>$count]);
+        });
```

Comment:
Ok for now, but there is a better way. You can declare [`sendPasswordResetNotification($token)`](https://laravel.com/api/5.8/Illuminate/Contracts/Auth/CanResetPassword.html) on user entity that will return the email. And it's better to create a separate email class that will contain all the logic of creation.

Reply:
I agree with you. But for fast delivery can we accept this variant? I created new task "Create custom email". And made it later.

Reply:
Ok